### ISSUE TITLE
`conda update --all` downgrades explicitly installed packages to update dependencies

### ISSUE THREAD

**Issue by lsorber** [conda/conda#4427]
TLDR; Conda will downgrade explicitly installed packages in order to update implicitly installed packages (dependencies of the explicitly installed packages).

Steps to reproduce this behaviour: I have a fresh install of Miniconda3 on my system with the following in my `.condarc`:
```
channels:
  - conda-forge
  - defaults
channel_priority: false
```
After the fresh install, I run:
1. `conda update --all` (resulting in conda version 4.3.8)
2. `conda install flake8 awscli`
3. `conda update --all`

On step (3) I see:
```
>conda update --all
Fetching package metadata .............
Solving package specifications: .

Package plan for installation in environment C:\Users\laurent\Miniconda3:

The following packages will be UPDATED:

    pyflakes: 1.3.0-py35_0 conda-forge --> 1.5.0-py35_0 conda-forge

The following packages will be DOWNGRADED due to dependency conflicts:

    flake8:   3.2.1-py35_0 conda-forge --> 3.0.4-py35_0 conda-forge

Proceed ([y]/n)?
```
Here, conda chooses to update the dependency `pyflakes`, causing the explicitly installed `flake8` to receive a significant downgrade. Instead, I would prefer it to not upgrade dependencies at the cost of downgrading explicitly installed packages (unless perhaps the number of updated packages outweighs the number of downgraded ones, but that is not the case here).

**Comment by nehaljwani** [comment]
I suspect this is happening because `conda info flake8` gives:

```bash
flake8 3.2.1 py35_0
-------------------
file name   : flake8-3.2.1-py35_0.tar.bz2
name        : flake8
version     : 3.2.1
build string: py35_0
build number: 0
channel     : conda-forge
size        : 838 KB
arch        : x86_64
has_prefix  : True
license     : MIT
license_family: MIT
md5         : 5e26865521d61875062070305934c2e4
platform    : linux
requires    : ()
subdir      : linux-64
url         : https://conda.anaconda.org/conda-forge/linux-64/flake8-3.2.1-py35_0.tar.bz2
dependencies:
    mccabe >=0.5,<0.6
    pycodestyle >=2.0,<2.3
    pyflakes >=1.2.3,<1.4
    python 3.5*
    setuptools
```

See the dependency on `pyflakes`. It has to be less than `1.4` and greater or equal to `1.2.3`. Since `conda update --all`  wants to update pyflakes from `1.3` to `1.5`, you see that error.

**Comment by kalefranz** [comment]
> Conda will downgrade explicitly installed packages in order to update implicitly installed packages (dependencies of the explicitly installed packages)

Indeed, it will, with the `--all` flag.  We don't right now incorporate knowledge of explicitly installed packages into update `--all`, but I agree it's something we should do.  Along with https://github.com/conda/conda/issues/4394

**Comment by mcg1969** [comment]
I'm in full agreement this would be a great addition, and indeed I've been bitten by this recently myself. We have to do a slightly better job of tracking an environment's conda install/remove history than we're currently doing, and then we can upgrade the solver to accomplish this.

**Comment by lsorber** [comment]
Glad to hear it! What would be the current recommended way to achieve a `conda update --all` without downgrading explicitly installed packages? Intuitively, I tried `conda update --all flake8=3.2`, but that didn't work.
